lib/cocina/generator/generator.rb

```diff
@@ -100,9 +100,13 @@ def schema_for(schema_name)
 
         case schema_doc.type
         when 'object'
-          Schema.new(schema_doc)
+          Schema.new(schema_doc, schemas: schemas.keys)
         when 'string'
-          Datatype.new(schema_doc)
+          Datatype.new(schema_doc, schemas: schemas.keys)
+        when NilClass
+          return unless schema_doc.one_of.map(&:name).all? { |ref_schema_name| schemas.keys.include?(ref_schema_name) }
+
+          UnionType.new(schema_doc)
```

Comment:
This is what allows us to now generate Rubyland models for `Barcode` and `DOI`, for example.